### PR TITLE
Test ScatterWidget 2D

### LINKED ISSUE #129: Fix features_scatter_widget unpredictable behaviour
In test_scatter, ScatterWidget was tested by adding 2 images layers and `viewer.layer.selection.add(viewer.layers[ind]` to select both images which appears to work as expected.

FeaturesScatterWidget was tested by adding an image, a labels and a points layers. The labels and a points layer both have associated features.
It is possible to plot feature_1 vs feature_0 for the points layer. However, both the labels and the points layer need to be selected for this to work, otherwise a blank figure is returned.
 
Another odd feature is that when adding the selection of the points and the label layers, the order is inconsistent and the FeaturesScatterWidget only recognises the first layer. 

So it appears that a 2nd layer needs to be selected (even if it's not used) so a blank figure is not returned and the FeaturesScatter plot only uses features in the 1st selected layer.

### PR DESCRIPTION
**Pull request by ruaridhg**
Towards #94 
- test_scatter.py: Added astronaut image and an inverted *-1 astronaut image to the viewer and made sure both image layers were selected before calling the ScatterWidget
- test_scatter.png added
- scatter.py: Added __init__ to ScatterWidget class to allow self.updates_layers(None) to add the selected layers (shouldn't have an effect on other parts of the code?)

For ScatterWidget 2D only (still to test 3D data and FeaturesScatterWidget)
Fixes #129

### CODE REVIEW

**Comment by dstansby** [comment]
It looks like you're tackling two different tests in this PR (scatter and features scatter)? Iis one of them working and one of them not? If one of them is working, I'd suggest putting that in a separate PR that we can merge straight away. Then we can keep moving, and it will make this PR easier to debug as it will only be concerned with one test.

**Comment by ruaridhg** [comment]
#137 solved inconsistent ordering problem. Partially addresses #129 with other behaviour perhaps desired

**Review by dstansby — CHANGES_REQUESTED** [review]
Looks 👍 overall, but we want to keep the number of layers features scatter takes as input (1), and plot two different features stored on that one layer against each other.

File: src/napari_matplotlib/scatter.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ class FeaturesScatterWidget(ScatterBaseWidget):
     Widget to scatter data stored in two layer feature attributes.
     """
 
-    n_layers_input = Interval(1, 1)
+    n_layers_input = Interval(2, 2)
```
[the remark this answers is not in the document]
Comment (ruaridhg) [reply src/napari_matplotlib/scatter.py:107]: Okay good to know, it already does that anyway so #129 should be mostly addressed. Just need to check if a 2nd layer needs to be added (even if not needed)

[the remark this answers is not in the document]
Comment (ruaridhg) [reply src/napari_matplotlib/scatter.py]: Only issue remaining is selecting two features in one layer (currently selects the same feature twice to plot against itself)

File: src/napari_matplotlib/tests/test_scatter.py [hunk 2]
```diff
@@ -25,16 +26,64 @@ def test_scatter(make_napari_viewer, astronaut_data):
...
+    widget.x_axis_key = x_column
+    y_column = "feature_1"
+    widget.y_axis_key = y_column  # features need to be set here
+
+    # Select label data and chosen features
+    # viewer.layers.selection.add(viewer.layers["label+features"])
+    # x_column = "label"
+    # widget.x_axis_key = x_column
+    # y_column = "perimeter"
+    # widget.y_axis_key = y_column
```
Comment (dstansby) [inline src/napari_matplotlib/tests/test_scatter.py:82]: Any reason for this to stay?

**Review by dstansby — CHANGES_REQUESTED** [review]

File: src/napari_matplotlib/tests/test_scatter.py [hunk 3]
```diff
@@ -25,16 +25,47 @@ def test_scatter(make_napari_viewer, astronaut_data):
...
+    )  # images need to be selected
+    widget.x_axis_key = "feature_0"
+    widget.y_axis_key = "feature_1"
+
+    # Select label data and chosen features
+    # viewer.layers.selection.add(viewer.layers["label+features"])
+    # x_column = "label"
+    # widget.x_axis_key = x_column
+    # y_column = "perimeter"
+    # widget.y_axis_key = y_column
```
Comment (dstansby) [inline src/napari_matplotlib/tests/test_scatter.py:64]:
```suggestion
```

File: src/napari_matplotlib/tests/test_scatter.py [hunk 4]
```diff
@@ -25,16 +25,47 @@ def test_scatter(make_napari_viewer, astronaut_data):
...
     viewer.layers.selection.add(viewer.layers[0])
     viewer.layers.selection.add(viewer.layers[1])
-
     return deepcopy(fig)
 
 
-def test_features_scatter_widget(make_napari_viewer):
-    # Smoke test adding a features scatter widget
+@pytest.mark.mpl_image_compare
+def test_features_scatter_widget_2D(make_napari_viewer, astronaut_data):
```
Comment (dstansby) [inline src/napari_matplotlib/tests/test_scatter.py:32]:
```suggestion
def test_features_scatter_widget_2D(make_napari_viewer):
```

File: src/napari_matplotlib/tests/test_scatter.py [hunk 5]
```diff
@@ -25,16 +25,47 @@ def test_scatter(make_napari_viewer, astronaut_data):
...
     viewer = make_napari_viewer()
-    viewer.add_image(np.random.random((100, 100)))
-    viewer.add_labels(np.random.randint(0, 5, (100, 100)))
-    FeaturesScatterWidget(viewer)
+    widget = FeaturesScatterWidget(viewer)
+
+    viewer.add_image(astronaut_data[0], **astronaut_data[1], name="astronaut")
```
Comment (dstansby) [inline src/napari_matplotlib/tests/test_scatter.py:36]:
```suggestion
```

**Review by dstansby — CHANGES_REQUESTED** [review]
I think there's still code that can be removed here (see my suggestions)?

**Comment by dstansby** [comment]
One final request (sorry!), could you squash this all into one commit so we don't add any more test figures changes than we need to in the commit history when it's merged into main?

**Comment by ruaridhg** [comment]
> 

1 last approval hopefully!

**Review by dstansby — APPROVED** [review]

**Comment by dstansby** [comment]
:shipit:

**Pull request merged**